src/components/SearchBox.tsx

```diff
@@ -73,36 +74,74 @@ export default function SearchBox({
     if (expanded && inputRef.current !== null) inputRef.current.select();
   }, [expanded]);
 
-  const searchLower = searchString.toLowerCase();
+  const searchLower = searchString.toLowerCase().split(/-| |,|_/);
+
+  const sortedRooms = useMemo(
+    () =>
+      data.rooms
+        .map((room) => ({
+          room,
+          number: parseFloat(room.number?.content || "0"),
+        }))
+        .sort((a, b) => (a.number === undefined ? 2 : a.number - b.number))
+        .map((room) => room.room),
+    [data.rooms],
+  );
+
+  const sortedBuildings = useMemo(
+    () =>
+      data.buildings.sort((a, b) => ((a.name || "") > (b.name || "") ? 1 : -1)),
+    [data.buildings],
+  );
+
+  const buildings =
+    searchString.length === 0
+      ? sortedBuildings
+      : sortedBuildings.filter(
+          (building) =>
+            building.name !== null &&
+            searchLower.includes(building.name.toLowerCase()),
+        );
```

Comment:
need to check for substrings

```suggestion
  const buildings =
    searchString.length === 0
      ? sortedBuildings
      : sortedBuildings.filter(
          (building) => {
            if (building.name === null) return false;
            const buildingNameLower = building.name.toLowerCase();
            return searchLower.some(str => buildingNameLower.includes(str))
        });
```